engine/src/main/java/org/terasology/world/block/family/MultiConnectFamily.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.block.family;
+
+import com.google.common.collect.Sets;
+import gnu.trove.map.TByteObjectMap;
+import gnu.trove.map.hash.TByteObjectHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.math.Rotation;
+import org.terasology.math.Side;
+import org.terasology.math.SideBitFlag;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.naming.Name;
+import org.terasology.registry.In;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
+import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
+
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Multi-Connect family describes a block family that will connect to other neighboring blocks.
```

Comment:
I can understand what this is about - I like it! Can we have such descriptions (with examples) for the other families as well, please?